openedx_learning/core/components/readme.rst

```diff
@@ -0,0 +1,22 @@
+Components App
+==============
+
+The ``components`` app holds the versioned data models for the lowest-level pieces of content that can be stored in Open edX: Components (e.g. XBlocks), as well as the individual pieces of raw data content that they reference.
+
+Motivation
+----------
+
+We want a small, extensible model for modeling the smallest pieces of content (e.g. individual blocks of XBlock content), that we will build more complex data on top of, like tagging.
+
+Intended Use Cases
+------------------
+
+
+
+Architecture Guidelines
+-----------------------
+
+* We're keeping nearly unlimited history, so per-version metadata (i.e. the space/time cost of making a new version) must be kept low.
```

Comment:
I don't suppose there's any way to tell which versions are in use or not with this model, because any application may or may not have foreign keys to specific `ComponentVersion`s?

(I'm wondering if we can auto-prune unused versions.)

Reply:
I think we can using Django ORM introspection or a mixin, but it wouldn't be perfect. I'm honestly hoping the per-version overhead is low enough now where it's okay not to prune, because that just makes content lifecycle management so much easier.